From 4c1da44a20a073fa5f1631cd382a8acf848c8ada Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Wed, 11 Dec 2013 01:47:10 +0000
Subject: [PATCH 0711/1115] ARM: OMAP2+: wkup_m3: Use reset framework to reset
 m3

Use the reset framework to deassert hard reset line on wkup_m3

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
Acked-by: Russ Dill <russ.dill@ti.com>
(cherry picked from commit 188559ea5fea47734a0fd5d2ed5dc9799f9fff83)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/wkup_m3.c |   16 ++++++++++++----
 1 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/wkup_m3.c b/arch/arm/mach-omap2/wkup_m3.c
index 2149e86..0765a4b 100644
--- a/arch/arm/mach-omap2/wkup_m3.c
+++ b/arch/arm/mach-omap2/wkup_m3.c
@@ -27,9 +27,8 @@
 #include <linux/interrupt.h>
 #include <linux/of.h>
 #include <linux/omap-mailbox.h>
-
-#include "pm33xx.h"
-#include "omap_device.h"
+#include <linux/reset.h>
+#include "wkup_m3.h"
 
 #define WKUP_M3_WAKE_SRC_MASK			0xFF
 
@@ -270,6 +269,7 @@ static irqreturn_t wkup_m3_txev_handler(int irq, void *unused)
 int wkup_m3_prepare(void)
 {
 	int ret = 0;
+	struct reset_control *rst_ctrl;
 	struct platform_device *pdev = to_platform_device(wkup_m3->dev);
 
 	wkup_m3->mbox = omap_mbox_get("wkup_m3", NULL);
@@ -283,7 +283,15 @@ int wkup_m3_prepare(void)
 	wkup_m3_fw_version_clear();
 
 	/* check that the code is loaded */
-	ret = omap_device_deassert_hardreset(pdev, "wkup_m3");
+	rst_ctrl = reset_control_get(&pdev->dev, NULL);
+
+	if (IS_ERR(rst_ctrl)) {
+		dev_err(wkup_m3->dev, "Unable to get reset control\n");
+		return -EINVAL;
+	}
+
+	ret = reset_control_deassert(rst_ctrl);
+	reset_control_put(rst_ctrl);
 
 	return ret;
 }
-- 
1.7.5.4

